client/txn.go

```diff
@@ -289,6 +279,25 @@ func (txn *Txn) Run(b *Batch) error {
 	return sendAndFill(txn.send, b)
 }
 
+func (txn *Txn) commit() error {
+	return txn.sendEndTxnCall(true /* commit */)
+}
+
+func (txn *Txn) cleanup(err error) {
+	if err != nil {
+		if replyErr := txn.Rollback(); replyErr != nil {
+			log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+		}
+	}
+}
+
+// CommitNoCleanup is the same as Commit but will not attempt to clean
+// up on failure. It is exposed only for use in txn_correctness_test.go
+// because those tests manipulate transaction state at a low level.
+func (txn *Txn) CommitNoCleanup() error {
```

Comment:
Not sure why my comment was ignored here. 

With fresh eyes, I'm still not liking `CommitNoCleanup`. It requires the caller to know how the txn was created. I think the `execing` flag nicely hid that from the caller. Adding a tiny bit of internal complexity to simplify an interface is well worth it.

And I'm going to harp on the dangers of cleanups one more time and then let it lie: cleanups are undeniably useful, but they come at a cost and with risks. A cost in the time it takes to write the code and the time it takes to review it. A risk that a new bug is introduced that isn't caught by existing tests or that a performance issue is created that wasn't covered by tests or benchmarks. The benefit of a cleanup is more maintainable/readable code, but that benefit does not always outweigh the costs and risks. I'm not arguing for a moratorium on cleanups, but for somewhat more consideration of the costs and risks.

That said, other than the introduction of `CommitNoCleanup`, this does like a good simplification.